parser.c

```diff
@@ -203,7 +203,7 @@ static char *get_optstr(char *buf, char string[])
 	len = strnlen(string, BUFSIZE);
 	sprintf(search_str, "%s=%%%ds\\n", string, BUFSIZE - len-1);
 	if (1 == sscanf(line, search_str, &temp)) {
-		len = strnlen(temp, 4096);
+		len = strnlen(temp, 4095) + 1;
```

Comment:
might want to drop your S-o-b line. as this change was created by Yan, Zheng in https://github.com/ceph/ceph/pull/22718/commits/843cddaaf83abdff0d900af85e8bd2382d834e0d

Reply:
Sure, will fix it.